nav2_loopback_sim/nav2_loopback_sim/loopback_simulator.py

```diff
@@ -64,6 +72,14 @@ def __init__(self):
         self.declare_parameter('scan_frame_id', 'base_scan')
         self.scan_frame_id = self.get_parameter('scan_frame_id').get_parameter_value().string_value
 
+        self.declare_parameter('publish_map_odom_tf', False)
```

Comment:
I don't understand this parameter. Why? 

At the very least, it should be default to true

Reply:
> I think a parameter for publishing map to odom will be the solution for this

I don't understand the take away. I'm saying that we shouldn't use TF buffer lookups. We have `self.t_map_to_odom` and `self.t_odom_to_base_link`. If trying to find the laser pose, we only need to find on initialization the `self.t_base_link_to_laser_scanner` and we have all the exact transforms needed to find the laser scanner link in the map frame without involving TF

Reply:
In latest commit I used the cached tf but still the same problem of flickering is there, when this parameter is set to False then it work well with nav2(no flickering)

**Both tested with LineIterator and with using cached tf**

https://github.com/user-attachments/assets/84a90c31-5bb2-49b2-8487-f619b665864b


https://github.com/user-attachments/assets/1f2aa291-0389-46ff-a17e-6c70b325b63e

Reply:
Is that happening before your changes? You're also showign a particle cloud, so I don't know what you mean by true/false because they're both using AMCL. This replaces AMCL when set to true -- maybe that's why you have flickering. You're running both simultaneously.

Try using the provided loopback launch files I provide in `nav2_bringup` - which does not have AMCL.